scripts/controller.gd

```diff
@@ -117,6 +118,22 @@ func _on_timer_timeout():
 	self._current_state = STATES.AWAITING_RESULT
 	self._board.turn_all_piece_lights_off()
 
+func _select_next_piece():
+	self._selected_piece_id = self._get_next_piece_id()
+	self._update_lights()
+
+func _get_next_piece_id() -> String:
+	var sorted = self._board.get_sorted_player_pieces()
+	if self._selected_piece_id == "":
+		return sorted[0]
```

Comment:
This will crash if the player no longer owns any pieces, won't it?
```suggestion
	var sorted = self._board.get_sorted_player_pieces()
	if sorted.is_empty():
	    return ""
	if self._selected_piece_id == "":
		return sorted[0]
```

Reply:
I'm mostly thinking about a game with > 2 players where one or more players have already lost all their pieces and just need to watch.